### PR TITLE
Create a database table

### PR DESCRIPTION
**Pull request by KAYEMBAJONATHAN**
- Add Create a database named vet clinic.
- Add Create a table named animals
- Add Copy the SQL used in the previous point into the data.sql file in the generated repository.
animals WHERE name NOT IN
animals where neutered
animals WHERE not weight
dateofbirth
Screenshot 2023-06-13 144013
Screenshot dataof birth where animals is
dateofbirth
animals WHERE not weight

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CREATE DATABASE vetclinic;
+
+CREATE TABLE animals (
+id: integer
+name: string
+date_of_birth: date
+escape_attempts: integer
+neutered: boolean
+weight_kg: integer
+);
```
Comment (Gambit142) [inline schema.sql]: - [ ] When I run this file, I get this error message:

image

#### Hint: This is not the correct syntax for creating table columns in Postgresql. You can read this [article](https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-create-table/#:~:text=The%20following%20statement%20creates%20the%20accounts%20table%3A) to learn how to create table columns the correct way. 👍 💻

Comment (Gambit142) [inline schema.sql]: - [ ] After creating the table columns the correct way, kindly make the `weight_kg` to be the data type `decimal` based on the project requirement. The decimal data type is for columns that will have values of the mathematical data type. 👍 😎

File: queries.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+select * FROM animals where Name like '%mon';
+SELECT * FROM animals WHERE date_of_birth BETWEEN '2016-01-01' AND '2019-12-31';
```
Comment (Gambit142) [inline queries.sql:2]: - [ ] Great work writing this query. However, you forgot to attach the screenshot of the query in your pull request. Please attach the screenshot as it helps verify that the correct results were returned from your database. 👍 😎 👍 

#### Hint: Attach the screenshot for this query:

```SQL
SELECT * FROM animals WHERE date_of_birth BETWEEN '2016-01-01' AND '2019-12-31';
```

File: queries.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+SELECT * FROM animals WHERE (neutered = true) AND (escape_attempts < 3);
```
Comment (Gambit142) [inline queries.sql:3]: - [ ] Although this query was well written, the screenshot for this query is missing. Please attach the screenshot as it helps verify that the correct results were returned from your database. 👍 😎 

#### Hint: Attach the screenshot for this query:

```SQL
SELECT * FROM animals WHERE (neutered = true) AND (escape_attempts < 3);
```

File: queries.sql [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+SELECT date_of_birth FROM animals WHERE NAME IN ( 'Agumon','Pikachu');
+SELECT name, escape_attempts FROM animals WHERE (weight_kg > 10.5);
```
Comment (Gambit142) [inline queries.sql:5]: - [ ] Great work writing the fifth query. However, you forgot to attach the screenshot of the query in your pull request. Please attach the screenshot as it helps verify that the correct results were returned from your database. 👍 😎 👍 

#### Hint: Attach the screenshot for this query:

```SQL
SELECT name, escape_attempts FROM animals WHERE (weight_kg > 10.5);
```

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @KAYEMBAJONATHAN 👋, 

Good job so far!

## Highlights
- Everything that creates or changes the structure of the table was saved in the **_schema.sql_** file ✔️ 
- Everything that populates tables with data was saved in the **_data.sql_** file ✔️ 
- Everything that is a query that helps to analyze data was saved in the **_queries.sql_** file ✔️

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by KAYEMBAJONATHAN** [comment]
> * Add Create a database named vet clinic.
> * Add Create a table named animals
> * Add Copy the SQL used in the previous point into the data.sql file in the generated repository.
>   animals WHERE name NOT IN
>   animals where neutered
>   animals WHERE not weight
>   dateofbirth
>   Screenshot 2023-06-13 144013
>   Screenshot dataof birth where animals is
>   dateofbirth
>   animals WHERE not weight

Screenshot escapa-attempted
Screenshot neutered

File: queries.sql (hunk 2 above)
Comment (abdulhamiid) [inline queries.sql:2]: As requested by the previous reviewer and according to the [project requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_create_tables.md), please attach the screenshot for this query 🛠

```sql
SELECT * FROM animals WHERE date_of_birth BETWEEN '2016-01-01' AND '2019-12-31';
```
Your screenshot
image

The result should list `the name` of all animals born between 2016 and 2019. 🤓

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @KAYEMBAJONATHAN 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, I really appreciate the effort you put in! but there are still some which haven't been addressed yet.

almost-there

## Highlights 🌟

- ✔️ Create tables in SQL.
- ✔️ Insert and query data in SQL.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by KAYEMBAJONATHAN** [comment]
> > * Add Create a database named vet clinic.
> > * Add Create a table named animals
> > * Add Copy the SQL used in the previous point into the data.sql file in the generated repository.
> >   animals WHERE name NOT IN
> >   animals where neutered
> >   animals WHERE not weight
> >   dateofbirth
> >   Screenshot 2023-06-13 144013
> >   Screenshot dataof birth where animals is
> >   dateofbirth
> >   animals WHERE not weight
> 
> Screenshot escapa-attempted Screenshot neutered

Screenshot animals WHERE date_of_birth BETWEEN '2016-01-01' AND '2019-12-31

**Review by prolajumokeoni — APPROVED** [review]
Hi @KAYEMBAJONATHAN 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - All changes requested have been implemented 🥇  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**